src/components/Filters/QueryFilter/QueryFilter.tsx

```diff
@@ -0,0 +1,36 @@
+import { useSearchParams } from 'react-router-dom';
+
+export const QueryFilter = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('search') || '';
+
+  const handleSearchQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.value) {
+      searchParams.delete('search');
+    } else {
+      searchParams.set('search', e.target.value);
+    }
+
+    setSearchParams(searchParams);
```

Comment:
You can do it in `setSearchParams`. The same use as for functional `setState`, you can pass a callback, get previous params modify them and return new